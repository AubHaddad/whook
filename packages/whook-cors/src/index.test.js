import { wrapHandlerWithCORS, optionsWithCORS } from '.';

describe('wrapHandlerWithCORS', () => {
  it('should work', async () => {
    const wrappedOptionsWithCORS = wrapHandlerWithCORS(optionsWithCORS);
    const handler = await wrappedOptionsWithCORS({
      CORS: {},
    });
    const response = await handler();

    expect({
      response,
    }).toMatchSnapshot();
  });
});
