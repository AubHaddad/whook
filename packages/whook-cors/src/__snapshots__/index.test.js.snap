// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`augmentAPIWithCORS() should work 1`] = `
Object {
  "basePath": "/v1",
  "host": "localhost:1337",
  "info": Object {
    "description": "A sample Swagger file for testing purpose.",
    "title": "Sample Swagger",
    "version": "1.0.0",
  },
  "paths": Object {
    "/ping": Object {
      "head": Object {
        "operationId": "ping",
        "responses": Object {
          "200": Object {
            "description": "Pong",
          },
        },
        "summary": "Checks API's availability.",
      },
      "options": Object {
        "consumes": Array [],
        "operationId": "optionsWithCORS",
        "parameters": Array [],
        "produces": Array [],
        "responses": Object {
          "200": Object {
            "description": "CORS sent.",
          },
        },
        "summary": "Enable OPTIONS for CORS",
        "x-whook": Object {
          "sourceOperationId": "ping",
          "suffix": "CORS",
          "type": "http",
        },
      },
    },
  },
  "schemes": Array [
    "http",
  ],
  "swagger": "2.0",
}
`;

exports[`wrapHandlerWithCORS should add CORS to errors 1`] = `
Object {
  "headers": Object {
    "Access-Control-Allow-Headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
    "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
    "Access-Control-Allow-Origin": "*",
    "Vary": "Origin",
  },
}
`;

exports[`wrapHandlerWithCORS should work 1`] = `
Object {
  "response": Object {
    "headers": Object {
      "Access-Control-Allow-Headers": "Accept,Accept-Encoding,Accept-Language,Referrer,Content-Type,Content-Encoding,Authorization,Keep-Alive,User-Agent",
      "Access-Control-Allow-Methods": "GET,POST,PUT,DELETE,OPTIONS",
      "Access-Control-Allow-Origin": "*",
      "Vary": "Origin",
    },
    "status": 200,
  },
}
`;
